File: Day 19 - Interfaces/Solution.java
import java.util.Scanner;

interface AdvancedArithmetic {
    int divisorSum(int n);
}

class Calculator implements AdvancedArithmetic{

    @Override
    public int divisorSum(int n){
        int sum = 0;
        for(int i = 1; i <=n; i++){
            if(n%i == 0) sum +=i;
        }
        return sum;
    }
}

class Solution{

    public static void main(String[] args){
        int n = new Scanner(System.in).nextInt();

        AdvancedArithmetic myCalculator = new Calculator();
        int sum = myCalculator.divisorSum(n);
        System.out.println("I implemented: " + myCalculator.getClass().getInterfaces()[0].getName() );
        System.out.println(sum);
    }
}

File: Day 19 - Interfaces/solution.php
<?php

interface AdvancedArithmetic
{
    public function divisorSum($n);
}

class Calculator implements AdvancedArithmetic{
    function divisorSum($n){
        $res = [];
        for($i = 1; $i <= $n; $i++){
            if($n % $i === 0){
                $res[] = $i;
            }
        }
        return array_sum($res);
    }
}

$n = intval(fgets(STDIN));
$myCalculator = new Calculator();
if ($myCalculator instanceof AdvancedArithmetic) {
    $sum = $myCalculator->divisorSum($n);
    echo 'I implemented: AdvancedArithmetic' . PHP_EOL . $sum;
}else{
    echo 'Wrong answer';
}
?>